utility/lambdas/ecs-asg-sizer/lambda_function.py

```diff
@@ -210,22 +224,62 @@ def get_usages(cluster_name):
     instance_arns = instance_list['containerInstanceArns']
     details = ecs_client.describe_container_instances(
         cluster=cluster_name, containerInstances=instance_arns)
-    return list(map(get_usage, details['containerInstances']))
+    usages = list(map(get_usage, details['containerInstances']))
+
+    # mark elapsed time agents have been disconnected
+    cluster_arn = get_cluster_arn(cluster_name)
+    disconnects = get_disconnects(cluster_arn)
+    now = int(datetime.datetime.now().timestamp())
+    for usage in usages:
+        if usage['disconnected'] and disconnects.get(usage['id']):
```

Comment:
Interesting, is there not an id in some cases?  Some properties of the cluster flickering?  I'm curious if this or other props indicate underlying state.  May be too hard to tell.

Reply:
I can do a better job commenting this as well.  The cases are:

1. This instance is connected.
2. This instance is disconnected for the first time ... so `disconnectseconds` is 0.
3. This instance is disconnected for the 2nd+ time ... need to calculate the elapsed `disconnectedseconds` using the current "now" epoch.